XML/TestCases/FunctionalTests-Power.xml

```diff
@@ -55,6 +55,44 @@
       <OverrideVMSize>Standard_D8s_v3</OverrideVMSize>
     </SetupConfig>
   </test>
+  <test>
+    <testName>POWER-HIBERNATE-SRIOV-DEALLOCATE</testName>
+    <testScript>Power-Hibernate.ps1</testScript>
+    <files>.\Testscripts\Linux\utils.sh,.\Testscripts\Linux\SetupHbKernel.sh</files>
+    <TestParameters>
+      <param>hb_url=HB_CUSTOM_KERNEL_URL</param>
+      <param>hb_branch=HB_CUSTOM_KERNEL_BRANCH</param>
+      <param>deallocate_vm=True</param>
+    </TestParameters>
+    <Platform>Azure</Platform>
+    <Category>Functional</Category>
+    <Area>POWER</Area>
+    <Tags>power,sriov</Tags>
+    <Priority>3</Priority>
+    <SetupConfig>
+      <Networking>SRIOV</Networking>
+      <SetupType>OneVM</SetupType>
+      <OverrideVMSize>Standard_D8s_v3</OverrideVMSize>
+    </SetupConfig>
+  </test>
+  <test>
+    <testName>POWER-HIBERNATE-GPU</testName>
```

Comment:
For this GPU VM hibernation, should we install GRID driver before the hibernation?

Reply:
Ref: 
https://docs.microsoft.com/en-us/azure/virtual-machines/linux/n-series-driver-setup
https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/hpccompute-gpu-linux
https://www.nvidia.com/Download/driverResults.aspx/150868/en-us
GRID version supporting Azure Linux: GRID: "430.30", "418.70", "410.92", "410.71", "390.75", "390.57", "390.42"

Reply:
> For this GPU VM hibernation, should we install GRID driver before the hibernation?

Yes. Before suspending, we should verify that the GPU driver works in the VM, e.g. 'nvidia-smi' shows the expected output, and /proc/interupts shows the GPU MSI interrupts are happening (i.e. the MSI interrupt counter keeps increasing). After resuming, we should verify the same thing.

Reply:
Thanks @juhlee-microsoft and @dcui ! I'll add these steps.

Reply:
Hi @juhlee-microsoft @dcui ,
I've just updated this PR to add the gpu driver installation and MSI interrupt value verification part. Please help to review again. Thanks!
This case will fail because the MSI interrupt stop increasing after waking up. It's similar with (https://bugzilla.redhat.com/show_bug.cgi?id=1876527 [Azure][RHEL-8]TX/RX packets stop increasing after hibernation/resume in VM with CX4 VF NIC) but I'm not quite sure if they are the same issue. @dcui Could you please help to have a look? Thanks!

**Test results:**
NV6(GRID driver)
```
09/29/2020 04:06:04 : [ERROR] Firsh check, MSI interrupts of nvidia driver - 501. Second check, MSI interrupts of nvidia driver - 501
09/29/2020 04:06:04 : [ERROR] EXCEPTION : GPU MSI interrupt stop increasing after waking up. at line: 447
```
```
[LISAv2 Test Results Summary]
Test Run On           : 09/29/2020 03:37:13
Total Test Cases      : 1 (0 Passed, 1 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:30

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 POWER                POWER-HIBERNATE-GPU                                                               FAIL                27.09
      OsVHD: http://walaautol2.blob.core.windows.net/vhds/RHEL-8.3.0-20200909.1.vhd, OverrideVMSize: Standard_NV6, TestLocation: eastus2, VMGeneration: 1, Kernel Version: 4.18.0-235.el8.x86_64
      Using nVidia driver : GRID



Logs can be found at C:\Users\yuxisun\LISAv2-yuxisun\TestResults\2020-09-29-11-35-48-9885
```

ND6s(CUDA driver)
```
09/29/2020 03:57:40 : [ERROR] First check, MSI interrupts of nvidia driver - 188. Second check, MSI interrupts of nvidia driver - 188
09/29/2020 03:57:40 : [ERROR] EXCEPTION : GPU MSI interrupt stop increasing after waking up. at line: 447
```
```
[LISAv2 Test Results Summary]
Test Run On           : 09/29/2020 03:06:15
VHD Under Test        : https://walaautoeastus.blob.core.windows.net/vhds/RHEL-8.3.0-20200909.1.vhd
Override VM size Under Test  : Standard_ND6s
Total Test Cases      : 1 (0 Passed, 1 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:53

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 POWER                POWER-HIBERNATE-GPU                                                               FAIL                48.37
      OsVHD: http://walaautoeastus.blob.core.windows.net/vhds/RHEL-8.3.0-20200909.1.vhd, OverrideVMSize: Standard_ND6s, TestLocation: eastus, VMGeneration: 1, Kernel Version: 4.18.0-235.el8.x86_64
      Using nVidia driver : CUDA



Logs can be found at C:\Users\yuxisun\LISAv2-yuxisun\TestResults\2020-09-29-11-05-17-5689
```

Reply:
> Hi @juhlee-microsoft @dcui ,
> I've just updated this PR to add the gpu driver installation and MSI interrupt value verification part. Please help to review again. Thanks!
> This case will fail because the MSI interrupt stop increasing after waking up. It's similar with (https://bugzilla.redhat.com/show_bug.cgi?id=1876527 [Azure][RHEL-8]TX/RX packets stop increasing after hibernation/resume in VM with CX4 VF NIC) but I'm not quite sure if they are the same issue. @dcui Could you please help to have a look? Thanks!
> 
> **Test results:**
> NV6(GRID driver)
> 
> ```
> 09/29/2020 04:06:04 : [ERROR] Firsh check, MSI interrupts of nvidia driver - 501. Second check, MSI interrupts of nvidia driver - 501
> 09/29/2020 04:06:04 : [ERROR] EXCEPTION : GPU MSI interrupt stop increasing after waking up. at line: 447
> ```

After resuming, the GPU interrupt is no longer happending. This is https://bugzilla.redhat.com/show_bug.cgi?id=1846838#c30
(The fix is under review: https://lkml.org/lkml/2020/9/8/1701)

Reply:
Got it. Thanks @dcui !